packages/apero-core/apero-core.0.4.7/opam

```diff
@@ -0,0 +1,45 @@
+opam-version: "2.0"
+version:"0.4.7"
+maintainer:   ["contact@adlink-labs.tech"]
+authors:      ["Angelo Corsaro" "Julien Enoch" "Oliver Hecart" "Gabriele Baldoni"]
+homepage:     "https://github.com/atolab"
+bug-reports:  "https://github.com/atolab/apero-core/issues/"
+dev-repo:     "git+https://github.com/atolab/apero-core.git"
+
+license: "Apache-2.0"
+
+
+build: [
+ ["dune" "build" "--only" "apero-core" "--root" "." "-j" jobs "@install"]
+]
+
+depends: [
+  "ocaml" {>= "4.03.0"}
+  "dune" {= "2.3.0" }
+  "astring" {= "0.8.3" }
+  "cmdliner" {= "1.0.4" }
+  "fmt"  {= "0.8.8" }
+  "logs"  {= "0.7.0" }
+  "lwt" {= "5.1.1" }
+  "ppx_deriving"  {= "4.4" }
+  "uuidm" {= "0.9.7" }
+  "bigstringaf"  {= "0.6.1" }
+  "lwt_ppx"  {= "2.0.0" }
+  "yojson"  {= "1.7.0" }
+  "ppx_deriving_yojson"  {= "3.5.1" }
```

Comment:
Is there a reason for dependency constraints this strict? Our CI already checks when something breaks and strict constrains are very annoying for users

Reply:
2 reasons:

1. We wanted reproductible builds (similarly to what's achieved in Java using Maven). I.e.: be sure that the version 0.4.7 of the software always leads to the same set of codes and thus to the same binary.
2. apero-core is used in [Eclipse zenoh](https://projects.eclipse.org/projects/iot.zenoh). And the Eclipse rules force us to precisely (i.e. with version, sources...) identify each piece of software that is a dependency of an Eclipse project. The goal being to provide legal guarantees wrt. IP used by an Eclipse software.